cli/src/main.rs

```diff
@@ -259,9 +259,74 @@ fn make_subcommands() -> Vec<Command> {
         .arg(Arg::new("module1").help("first module ID or path to .wasm"))
         .arg(Arg::new("module2").help("second module ID or path to .wasm"));
 
-    [create, delete, get, list, search, validate, yank, audit]
-        .into_iter()
-        .map(add_output_arg)
-        .chain(vec![generate, diff])
-        .collect()
+    let call_plugin = clap::Command::new("call")
+        .about("Call a Modsurfer plugin.")
+        .arg(
+            Arg::new("identifier")
+                .long("id")
+                .help("the identifier of the registered plugin"),
+        )
+        .arg(
+            Arg::new("function")
+                .long("function")
+                .short('f')
+                .help("the function to be called"),
+        )
+        .arg(
+            Arg::new("input")
+                .long("input")
+                .short('i')
+                .help("use @{path_to_file} to specify a file as input to your plugin's function. Otherwise, the value provided will used as input to your function as raw bytes"),
+        )
+        .arg(
+            Arg::new("output")
+                .value_parser(clap::value_parser!(PathBuf))
+                .long("output")
+                .short('o')
+                .help("a location on disk to write the output.  The output of the call will be written to stdout if not specified"),
+        );
+
+    // TODO: allow specification of plugin "config"
+    let install_plugin = clap::Command::new("install")
+        .about("Install a Modsurfer plugin.")
+        .arg(
+            Arg::new("identifier")
+                .long("id")
+                .help("the identifier of the plugin to be installed"),
+        )
+        .arg(
+            Arg::new("name")
+                .required(false)
+                .long("name")
+                .short('n')
+                .help("the human readable name of the plugin"),
+        )
+        .arg(
+            Arg::new("wasm")
+                .long("wasm")
+                .short('w')
+                .help("a path on disk to the plugin"),
+        );
+
+    let uninstall_plugin = clap::Command::new("uninstall")
+        .about("Uninstall a Modsurfer plugin.")
+        .arg(
+            Arg::new("identifier")
+                .long("id")
+                .help("the identifier of the plugin to uninstall"),
+        );
+
+    let plugin = clap::Command::new("plugin")
+        .about("Manage your Modsurfer plugins")
+        .subcommand(call_plugin)
+        .subcommand(install_plugin)
+        .subcommand(uninstall_plugin);
+
+    [
+        create, delete, get, list, search, validate, yank, audit, plugin,
```

Comment:
`plugin` should be added to the `.chain(vec![...])` instead, since the results from the plugin call can't be formatted via JSON/Table reliably. Sorry there's no obvious documentation for this..

Reply:
:+1: good catch!